source/slang/slang-compiler.cpp

```diff
@@ -1193,24 +1193,46 @@ void printDiagnosticArg(StringBuilder& sb, CodeGenTarget val)
         // Disable exceptions and security checks
         options.flags &= ~(CompileOptions::Flag::EnableExceptionHandling | CompileOptions::Flag::EnableSecurityChecks);
 
+        Profile profile;
+
         if (compilerType == PassThroughMode::Fxc ||
             compilerType == PassThroughMode::Dxc ||
             compilerType == PassThroughMode::Glslang)
         {
-            auto entryPointIndex = getSingleEntryPointIndex();
+            const auto entryPointIndices = getEntryPointIndices();
+            auto targetReq = getTargetReq();
 
-            auto entryPoint = getEntryPoint(entryPointIndex);
-            auto profile = getEffectiveProfile(entryPoint, getTargetReq());
+            const auto entryPointIndicesCount = entryPointIndices.getCount();
+             
+            if (entryPointIndicesCount == 0 && compilerType == PassThroughMode::Dxc)
+            {
+                // Can support no entry points on DXC because we can build libraries
+                profile = targetReq->getTargetProfile();
+            }
+            else if (entryPointIndicesCount == 1)
+            {
```

Comment:
This is reasonable for now, but I think we discussed separately how it would be good to change the `CodeGenContext::EntryPointIndices` type so that we can detect whether are are in "single entry point mode" or in "library mode" (aka "whole program" mode in the current naming). We ideally want to allow "whole program mode" compilation for DXIL even when there are one or more entry points.

(Note that at this point in compilation the automatic detection of entry points has already happened in the Slang front-end, so we might have `entryPointIndicesCount > 1` here even in "library mode," unless I'm missing something)

While the above would be the best long-term fix, it seems that in the short term this could at least be written to check `entryPointIndicesCount != 1` as a way to handle the 0 entry point and >1 entry point cases.

Reply:
I do use the 'whole program' mechanism in my later PRs.